### PR TITLE
Optimize network subsystem initialization

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- net: Remove net_lockinitialize
- net/local: Remove local_initialize 
- net/tcp: Remove tcp_listen_initialize
- net/route: Remove net_init_fileroute
- net/igmp: Remove igmp_initialize 
- net: Remove the unnecessary initialization code
- net: Reoder the initialize sequence(mac->ip->tcp/udp) 

## Impact
Minor, code refactor

## Testing
Pass CI

### CODE REVIEW

File: net/utils/net_lock.c [hunk 1]
```diff
@@ -49,9 +49,9 @@
  * Private Data
  ****************************************************************************/
 
-static sem_t        g_netlock;
-static pid_t        g_holder = NO_HOLDER;
-static unsigned int g_count  = 0;
+static sem_t        g_netlock = SEM_INITIALIZER(1);
+static pid_t        g_holder  = NO_HOLDER;
+static unsigned int g_count   = 0;
```
Comment (pkarashchenko) [inline net/utils/net_lock.c:54]:
```suggestion
static unsigned int g_count;
```

**Review by pkarashchenko — COMMENTED** [review]
Maybe we can remove
```
#define sq_init(q) \
  do \
    { \
      (q)->head = NULL; \
      (q)->tail = NULL; \
    } \
  while (0)

#define dq_init(q) \
  do \
    { \
      (q)->head = NULL; \
      (q)->tail = NULL; \
    } \
  while (0)
```
if those are not used anymore

File: net/utils/net_lock.c (hunk 1 above)
Comment (xiaoxiang781216) [reply net/utils/net_lock.c]: Done.

**Comment by xiaoxiang781216** [comment]
> Maybe we can remove
> 
> ```
> #define sq_init(q) \
>   do \
>     { \
>       (q)->head = NULL; \
>       (q)->tail = NULL; \
>     } \
>   while (0)
> 
> #define dq_init(q) \
>   do \
>     { \
>       (q)->head = NULL; \
>       (q)->tail = NULL; \
>     } \
>   while (0)
> ```
> 
> if those are not used anymore

Still has some caller, let's do in the new patch.

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
Please add `const` to `ip` parameter of `void igmp_removemcastmac(FAR struct net_driver_s *dev, FAR in_addr_t *ip)` and `static void igmp_mcastmac(in_addr_t *ip, FAR uint8_t *mac)`

**Comment by xiaoxiang781216** [comment]
Done, @pkarashchenko .

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**